spec/consensus/light-client.md

```diff
@@ -60,259 +61,372 @@ For the purpose of this lite client specification, we assume that the Tendermint
   }
-```
 
-_Remark_: Basic header verification must be done for _h2_. Similar checks are done in:  
- https://github.com/tendermint/tendermint/blob/master/types/validator_set.go#L591-L633
-
-_Remark_: There are some sanity checks which are not in the code:
-_h2.Header.height > h1.Header.height_ and _h2.Header.bfttime > h1.Header.bfttime_ and _h2.Header.bfttime < now_.
+  // return true if header is correctly signed by 2/3+ voting power in the corresponding validator set; 
+  // otherwise false. Additional checks should be done in the implementation
+  // to ensure header is well formed.
+  func verify(h) bool {
+    vp_all := totalVotingPower(h.Header.V) // total sum of voting power of validators in h
+    return votingpower_in(signers(h.Commit),h.Header.V) > 2/3 * vp_all
+  }
 
-_Remark_: `return (votingpower_in(signers(h2.Commit),h1.Header.NextV) > max(1/3,trustlevel) * vp_all)` may return false even if _h2_ was properly generated by Tendermint consensus in the case of big changes in the validator sets. However, the check `return (votingpower_in(signers(h2.Commit),h1.Header.NextV) > 2/3 * vp_all)` must return true if _h1_ and _h2_ were generated by Tendermint consensus.
+  // Captures skipping condition. h1 and h2 has already passed basic validation (function `verify`).
+  // returns nil in case h2 can be trusted based on h1, otherwise returns error.
+  // ErrHeaderExpired is used to signal that h1 has expired with respect lite client trusted period,
+  // ErrInvalidAdjacentHeaders that adjacent headers are not consistent and
+  // ErrTooMuchChange that there is not enough intersection between validator sets to have skipping condition true.
+  func CheckSupport(h1,h2,trustlevel) error {
+      assume h1.Header.Height < h2.header.Height and h1.Header.bfttime < h2.Header.bfttime and h2.Header.bfttime < now
+
+      if !isWithinTrustedPeriod(h1) return ErrHeaderNotWithinTrustedPeriod(h1)
+
+      // Although while executing the rest of CheckSupport function, h1 can expiry based on the lite client trusted period, this is not problem as
+      // lite client trusted period is smaller than trusted period of the header based on Tendermint Failure model, i.e., there is a significant
+      // time period (measure in days) during which validator set that has signed h1 can be trusted
+      // Furthermore, CheckSupport function is not doing expensive operation (neither rpc nor signature verification), so it should execute fast.
+
+      // total sum of voting power of validators in h1.NextV
+      vp_all := totalVotingPower(h1.Header.NextV)
+
+      // check for adjacent headers
+      if (h2.Header.height == h1.Header.height + 1) {
+            if h1.Header.NextV == h2.Header.V 
+                return nil
+            return ErrInvalidAdjacentHeaders
+      }
+      // check for non-adjacent headers
+      if votingpower_in(signers(h2.Commit),h1.Header.NextV) > max(1/3,trustlevel) * vp_all return nil
```

Comment:
That's not really related to the bisection changes but I'm wondering if `trustLevel` isn't a confusing name? Isn't behaving more like a threshold? How does `trustThreshold` sound instead?